soco/core.py

```diff
@@ -1833,6 +1873,185 @@ def add_item_to_sonos_playlist(self, queueable_item, sonos_playlist):
             ('AddAtIndex', 4294967295)
         ])
 
+    def reorder_sonos_playlist(self, object_id, tracks, new_pos, update_id=0):
+        """ Reorder and/or Remove tracks in a Sonos playlist. The underlying
+        call is quite complex as it both can move a track within the list or
+        delete a track from the playlist.  All of this depends on what tracks
+        and new_pos specify.
+
+        See the helper methods
+        :py:meth:`clear_sonos_playlist`, :py:meth:`move_in_sonos_playlist`,
+        :py:meth:`remove_from_sonos_playlist` for simplified usage.
+
+        update_id - If you have a series of operations, tracking the update_id
+        and setting it, will save a lookup operation.
+
+        Examples:
+          To reorder the first two tracks::
+
+            # object_id specified by the item_id
+            obj.reorder_sonos_playlist('SQ:0', tracks=[0, ], new_pos=[1, ])
+            # OR object_id specified by the DidlPlaylistContainer object
+            spl = obj.get_sonos_playlists()[0]
+            obj.reorder_sonos_playlist(spl, tracks=[0, ], new_pos=[1, ])
+
+          To delete the second track::
+
+            # tracks/new_pos are a list of int
+            obj.reorder_sonos_playlist('SQ:0', tracks=[1, ], new_pos=[None, ])
+            # OR tracks/new_pos are a list of int-like
+            obj.reorder_sonos_playlist('SQ:0', tracks=['1', ], new_pos=['', ])
+            # OR tracks/new_pos are strings - no transform is done
+            obj.reorder_sonos_playlist('SQ:0', tracks='1', new_pos='')
```

Comment:
Yes, but as I mentioned somewhere later, the reorder_sonos_playlist could (and maybe should) simply turn the iterable inputs into strings form, and always perform just one request.